### ISSUE TITLE
rpush 2.3.2 not working with redis

### ISSUE THREAD

**Issue by simonknoll** [rpush/rpush#125]
- I am using following classes for rpush gcm and redis
  - Rpush::Client::Redis::Gcm::App
  - Rpush::Client::Redis::Gcm::Notification
- in the rpush.rb initializer the following is set:
  - config.client = :redis
- I've added the rpush-redis gem
- when I start rpush with ´rpush start´ i get the following in the logs

```
[2015-02-02 15:59:35] [ERROR] NoMethodError, undefined method `service_name' for #<Rpush::Client::Redis::App:0x007f9b3ce70ea8>
/Users/simon/.rvm/gems/ruby-2.1.2/gems/activemodel-4.1.9/lib/active_model/attribute_methods.rb:435:in `method_missing'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/app_runner.rb:183:in `service'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/app_runner.rb:175:in `new_dispatcher_loop'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/app_runner.rb:101:in `block in start_dispatchers'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/app_runner.rb:101:in `times'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/app_runner.rb:101:in `start_dispatchers'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/app_runner.rb:31:in `start_app'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/synchronizer.rb:18:in `sync_app'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/synchronizer.rb:9:in `block in sync'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/synchronizer.rb:9:in `each'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon/synchronizer.rb:9:in `sync'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/daemon.rb:59:in `start'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/lib/rpush/cli.rb:26:in `start'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
/Users/simon/.rvm/gems/ruby-2.1.2/gems/rpush-2.3.2/bin/rpush:4:in `<top (required)>'
/Users/simon/.rvm/gems/ruby-2.1.2/bin/rpush:23:in `load'
/Users/simon/.rvm/gems/ruby-2.1.2/bin/rpush:23:in `<main>'
/Users/simon/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `eval'
/Users/simon/.rvm/gems/ruby-2.1.2/bin/ruby_executable_hooks:15:in `<main>'

```

**Comment by ileitch** [comment]
For some reason your app single-table inheritance `type` is `Rpush::Client::Redis::App`, where it should be `Rpush::Client::Redis::Gcm::App`.

Can you show me the code where you create your apps?

**Comment by simonknoll** [comment]
Hello @ileitch, thanks for the fast response, here the code:

```ruby
class PushNotifier
  ANDROID_APP_NAME = "android_app"
  def self.get_app
    app = Rpush::Client::Redis::Gcm::App.where(:name => ANDROID_APP_NAME)
    if app.blank?
      app = Rpush::Client::Redis::Gcm::App.new
      app.name = ANDROID_APP_NAME
      app.auth_key = ENV['gcm_server_api_key']
      app.connections = 1
      app.save!
    end
    return app
  end

  def self.send_push_notification token, message
    app = PushNotifier.get_app
    n = Rpush::Client::Redis::Gcm::Notification.new
    n.app = app
    n.registration_ids = [token]
    n.data = message
    return n.save!
  end

  def self.message context, state, id
    { context: context,
      content: {
        state: state,
        uuid: id.to_s
      }
    }
  end
end
```

**Comment by ileitch** [comment]
What version of [Modis](https://github.com/ileitch/modis) do you have installed?

What does the output of the following give you?

```ruby
Rpush::Gcm::App.all.each { |app| p [app.id, app.name, app.type] };nil
```

**Comment by ileitch** [comment]
@simonknoll any luck?

**Comment by aried3r** [comment]
We've just released rpush 3.0.0.rc1 and there's also been a few more modis and rpush-redis versions. Feel free to reopen if this is still an issue for you.
